### PR TITLE
feat(ci): run e2e test weekly

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

- Added an action to build image from a a specific branch and push it to gcr.io/zeebe-io repo. The tag follows SemVer so that it can be used in tests run via testbench. This action is mostly extracted from `qa_testbench` workflow. 
- Added a workflow to run weekly e2e tests. The workflow does not monitor the test results. Testbench notifies via slack channel if the test fails. Any incidents on the test process will be noticed by the medic.

### CODE REVIEW

File: .github/actions/build-push-docker-gcr/action.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        branch=${branch:-main}
+        echo "BRANCH_NAME=$branch" >> $GITHUB_ENV
+      env:
+        BRANCH: "${{ inputs.branch }}"
+    # We need to check out the evaluated branch and setup java (incl. maven), so we can retrieve the current project version
+    # The version is necessary, since CC Saas only accepts SemVer for docker image tags (need to start with a version tag)
+    - uses: actions/checkout@v3
+      with:
+        ref: "${{ github.event.inputs.branch }}"
+    - uses: actions/setup-java@v3.6.0
```
Comment (npepinpe) [inline .github/actions/build-push-docker-gcr/action.yml:47]: :x: `setup-zeebe` will set up Java. We can already call it here before we even generate the TAG. So, e.g. the ordering can be:

```yaml
   - uses: actions/checkout@v3
   - uses: hashicorp/vault-action@v2.4.2
   - uses: ./.github/actions/setup-zeebe
   - uses: ./.github/actions/build-zeebe
   - id: generate-tag
   - uses: docker/login-action@v2
   - uses: ./.github/actions/build-docker
```

File: .github/workflows/e2e-testbench.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          secrets: |
+            secret/data/products/zeebe/ci/zeebe TESTBENCH_PROD_CLIENT_SECRET;
+            secret/data/products/zeebe/ci/zeebe TESTBENCH_PROD_CONTACT_POINT;
+        # Start e2e test instance, do not wait for the result
+      - name: Run E2E Tests
+        run: .ci/scripts/distribution/e2e-testbench.sh
```
Comment (npepinpe) [inline .github/workflows/e2e-testbench.yaml:59]: :question: Is there any benefit from running this as a script and not just inline here? e.g.

```yaml
      - name: Run E2E Tests
        run: |
          chmod +x clients/go/cmd/zbctl/dist/zbctl
          read -r -d '' VARIABLES << EOM
          {
            "zeebeImage":"${{ steps.build-docker.outputs.image }}",
            "generationTemplate":"${{ github.event.inputs.generation || 'Zeebe SNAPSHOT' }}",
            "channel":"Internal Dev",
            "clusterPlan":"Production - M",
            "region":"new chaos",
            "properties":[
                "allInstancesAreCompleted"
            ],
            "testProcessId":"e2e-test",
            "branch":"${{ github.event.inputs.branch }}",
            "build":"${{ env.BUILD_URL }}",
            "testParams":{
                "maxTestDuration":"${{ github.event.inputs.maxTestDuration || 'PT5D' }}",
                "starter":[
                  {
                      "rate":50,
                      "processId":"one-task-one-timer"
                  }
                ],
                "verifier":{
                  "maxInstanceDuration":"15m"
                }
            }
          }
          EOM

          clients/go/cmd/zbctl/dist/zbctl create instance e2e_testbench_protocol --variables "${VARIABLES}"
```

Although, now I wonder if we just shouldn't have a `zbctl` action anyway.

File: .github/workflows/e2e-testbench.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        description: 'Test duration (Eg: PT2H, P3D)'
+        required: true
+        default: 'PT1D'
+        type: string
+      generation:
+        description: 'Specifies the generation template which should be used by the testbench run'
+        required: false
+        default: 'Zeebe SNAPSHOT'
+        type: string
+      branch:
```
Comment (npepinpe) [inline .github/workflows/e2e-testbench.yaml:16]: :x: I think you don't need the branch input, workflow_dispatch always has the branch or tag, which you access using `GITHUB_REF`. See https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch and https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_dispatch.

**Review by npepinpe — APPROVED** [review]
:+1: 

Only minor things, please see the comments, but I don't think it requires another review pass :+1:

File: .github/workflows/e2e-testbench.yaml (hunk 2 above)
Comment (oleschoenburg) [reply .github/workflows/e2e-testbench.yaml:59]: We have one :sweat_smile: https://github.com/zeebe-io/zbctl-action/

Comment (deepthidevaki) [reply .github/workflows/e2e-testbench.yaml]: What you proposed did not work. The script failed, but it didn't say why.  So I have just inlined the script here.

Comment (deepthidevaki) [reply .github/workflows/e2e-testbench.yaml]: > We have one sweat_smile https://github.com/zeebe-io/zbctl-action/

Nice. Totaly forgot about it. For now I wont use it, as zbctl is already available without the action.

**Comment by deepthidevaki** [comment]
bors merge

File: .github/workflows/e2e-testbench.yaml (hunk 3 above)
Comment (deepthidevaki) [reply .github/workflows/e2e-testbench.yaml]: :+1: used `github.ref_name`

**Pull request merged**